### PR TITLE
Configuring the plugin via params

### PR DESCRIPTION
**Pull request by lurecas**
As we are working on a Cordova (mainly Android) hybrid app and we want to integrate our product with Appboy, we began to develop a plugin to do so. But this came out yesterday and certainly saved our day.

We have been working a while with Cordova and some plugins and the only thing that we missed was the possibility of integrating the plugin on the app without having to fork/download it. 

Inspired on the configuration capabilities of this plugin : https://github.com/smistry-toushay/cordova-crashlytics-plugin , we've mimicked the mechanism to do so.

To install the plugin you now would do
```bash
cordova plugin add https://github.com/Appboy/appboy-cordova-sdk --variable APPBOY_API_KEY="$YOUR_API_KEY" --variable APPBOY_PUSH_REGISTRATION_ENABLED="false/true" --variable APPBOY_GCM_SENDER_ID="$YOUR_SENDER_ID"
```

Basically, we run a hook to detect when the plugin have been added to modify the base `appboy.xml` file with the values that the user inputs at install time. We will also edit the `$PACKAGE_NAME` on the `AppboyBroadcastReceiver.java` via the same hook.

Thanks!

### CODE REVIEW

**Comment by lurecas** [comment]
We only made it for the Android app because we are only currently working on that.

**Comment by briancaw** [comment]
Hey @lurecas thanks very much for this pull request!  We'll take a look today and get this merged ASAP.

**Comment by lurecas** [comment]
@briancaw We also add one function that we needed at the moment. Feel free to contact me if something is not working correctly

**Comment by briancaw** [comment]
Hey again @lurecas - do you know if there's a way to get cordova to recognize that the plugin is in a subdirectory?  It seems to want to find the plugin.xml in the root of the repo and is failing because it's not there.  i.e.  `cordova plugin add https://github.com/cabify/appboy-cordova-sdk --variable APPBOY_API_KEY="7ef72382-b92f-4258-9be9-9e19e92f3bf1" --variable APPBOY_PUSH_REGISTRATION_ENABLED=“true" --variable APPBOY_GCM_SENDER_ID=“7ef72382-b92f-4258-9be9-9e19e92f3bf1"` is failing.

No worries if not, I think it's probably simple enough to move the plugin or the plugin.xml up a directory.  Thanks.

**Comment by briancaw** [comment]
Edit @lurecas:  I just found it's possible via cordova plugin add https://github.com/cabify/appboy-cordova-sdk#:/plugin/ --variable APPBOY_API_KEY="7ef72382-b92f-4258-9be9-9e19e92f3bf1" --variable APPBOY_PUSH_REGISTRATION_ENABLED=“true" --variable APPBOY_GCM_SENDER_ID=“7ef72382-b92f-4258-9be9-9e19e92f3bf1"

File: plugin/src/android/AppboyPlugin.java [hunk 1]
```diff
@@ -10,6 +10,8 @@
 import com.appboy.enums.NotificationSubscriptionType;
 import com.appboy.models.outgoing.AppboyProperties;
 import com.appboy.ui.activities.AppboyFeedActivity;
+import com.appboy.models.outgoing.AttributionData;
+import com.adjust.sdk.AdjustAttribution;
```
Comment (briancaw) [inline plugin/src/android/AppboyPlugin.java]: This is throwing an error on a regular app because Adjust isn't a dependency of our SDK; also I don't believe you're using it, so I think it can just be removed?

File: plugin/www/AppboyPlugin.js [hunk 2]
```diff
@@ -93,6 +93,14 @@ AppboyPlugin.prototype.submitFeedback = function (email, feedback, isBug) {
 
 // Appboy user methods
 /**
+ * Sets the user Attribution Data. We are using this to connect our integration with Adjust
```
Comment (briancaw) [inline plugin/www/AppboyPlugin.js]: Can we modify this to remove the second sentence so that the method remains generic for attribution data.

**Comment by briancaw** [comment]
Hey @lurecas 

So I tested this out and it works great (once I comment out that adjust import line).  I have a few concerns that the node dependencies required for this to work (e.g. xpath, xmldom, cordova-lib) combined with the fact that they'd be required (i.e. there's no way I can see to turn off this requirement with the current state of the code) might make things less than ideal for some folks.

That said, I think there's two main options.  The first would be to modify things so that the parameters and libraries are optional.  However, I think there's downsides/complexity to that solution.   The second option I see is merging this into it's own branch and having the variable-passing integration method be accessed that way.  That's what I'm leaning towards.

I've created a branch on this repo (https://github.com/Appboy/appboy-cordova-sdk/tree/feature/android-variable-integration) called feature/android-variable-integration that I think we could use for this purpose.  If you modify the import and reopen the PR on that branch, we can merge that in and then the integration point would be:  `cordova plugin add https://github.com/appboy/appboy-cordova-sdk#feature/android-variable-integration:/plugin/ --variable APPBOY_GCM_SENDER_ID=SENDER_ID --variable APPBOY_API_KEY=API_KEY --variable APPBOY_PUSH_REGISTRATION_ENABLED=true\false` and we'll modify the readme to inform folks of this option.

If you have a different solution you think would work better, please let us know and we can work to accommodate that.  Thanks again very much for helping us improve this SDK.

- Brian

File: plugin/src/android/AppboyPlugin.java (hunk 1 above)
Comment (lurecas) [inline plugin/src/android/AppboyPlugin.java]: Wow, my bad. I keep on working and and a pair of commits got out of hand to this PR.
We are working with Appboy + Adjust, but I think I need to find a better way to handle with this.

File: plugin/www/AppboyPlugin.js (hunk 2 above)
Comment (lurecas) [inline plugin/www/AppboyPlugin.js]: Yeah, I wrote it too fast, and I even forgot to write the comments for the params. I will get to that, the comments on the other method are so helpful

**Comment by lurecas** [comment]
Hi @briancaw 

I didn't hesitate to add the node dependencies because they are on our current stack, and it surely feel handy. I will ask my coworkers so if we can use another thing to accomplish that. Maybe we can do it with plain JS dependencies, it was just a test to see if it was a viable option to install the plugin.

As I told you, I am only working on the Android side because it's the platform we are targeting now, but I think I will have some time to get to the iOS part.

I like the new branch thing, as I have some more things that I had to implement on the plugin that was missing from your work that we need. I will get to that tomorrow asap. And so I will try to decouple all the things related to Adjust as we are also using their Cordova SDK.

You wouldn't believe the timing of this repo coming out, I just began on Monday working on my own version (much much worse that yours) of a minimum plugin of the Appboy's functionalities we need on our Cordova app. Just on the middle of my working day a coworker said, " you are not going to believe what I just found on the Appboy Github repo". And here we are :smile: . Thank you, it would surely take me like a week to have all of this working!

**Comment by briancaw** [comment]
Sounds good @lurecas; very happy that we were able to save you some work.  Looking forward to your updates and to have your solution available to everyone - I think it will be a really nice and convenient integration for folks.  Thanks again!

**Comment by lurecas** [comment]
I've opened a new PR, https://github.com/Appboy/appboy-cordova-sdk/pull/2
